src/nowsecure-upload-app.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2021-2022 NowSecure Inc.
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+import * as core from "@actions/core";
+import fs from "fs";
+import { NowSecure } from "./nowsecure-client";
+import { outputToDependencies, outputToSarif } from "./utils/action-utils";
+import * as github from "@actions/github";
+
+async function run() {
+  try {
+    const apiUrl = core.getInput("api_url");
+    const labApiUrl = core.getInput("lab_api_url");
+    const labUrl = core.getInput("lab_url");
+    const platformToken = core.getInput("token");
+    const ns = new NowSecure(platformToken, apiUrl, labApiUrl);
+    const groupId = core.getInput("group_id");
+    const appFile = core.getInput("app_file");
+    const enableSarif = core.getBooleanInput("enable_sarif");
+    const enableDependencies = core.getBooleanInput("enable_dependencies");
+    const githubToken = core.getInput("github_token");
+    const githubCorrelator = core.getInput("github_correlator");
+
+    let reportId = core.getInput("report_id");
+
+    if (reportId) {
+      console.log(`Pulling report with ID: ${reportId}`);
+      const report = await ns.pullReport(reportId);
+
+      if (enableDependencies) {
+        await outputToDependencies(
+          report,
+          github.context,
+          githubCorrelator,
+          githubToken
+        );
+      }
+
+      if (enableSarif) {
+        await outputToSarif(report, labUrl);
+      }
+    } else {
+      // No report_id was passed in.
+      const details = await ns.submitBin(fs.createReadStream(appFile), groupId);
+      reportId = details.ref;
+      console.log(`NowSecure assessment started. Report ID: ${reportId}`);
+      core.setOutput("report_id", reportId);
+
+      return reportId;
+    }
+  } catch (e) {
+    console.error(e);
+    core.setFailed((e as Error).message);
+  }
+}
+
+run();
```

Comment:
not going to block for this, but maybe consideration as a follow up to move anything that's intended to be an action entry point we put in a specific folder